manta-crypto/src/hash.rs

```diff
@@ -26,17 +24,8 @@ pub trait UnaryHashFunction<COM = ()> {
     /// Output Type
     type Output;
 
-    /// Computes the hash over `input` in the given `compiler`.
-    fn hash_in(&self, input: &Self::Input, compiler: &mut COM) -> Self::Output;
-
     /// Computes the hash over `input`.
-    #[inline]
-    fn hash(&self, input: &Self::Input) -> Self::Output
-    where
-        COM: Native,
-    {
-        self.hash_in(input, &mut COM::compiler())
-    }
+    fn hash(&self, input: &Self::Input, compiler: &mut COM) -> Self::Output;
```

Comment:
I got rid of all the old `_owned` APIs because we have to wait on more ECLAIR design work before we can settle on whether or not ECLAIR can/should auto-generate those definitions. This is the same reason I got rid of the use of the `Native` marker. We have to do a few more iterations before it's clear what to do, and better to just have less code now if possible, we can grow it in the future.